recipes-app/switchserialmode/files/src/transceiver/sp339e.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Siemens AG, 2019-2022
+ *
+ * Authors:
+ *  Gao Nian <nian.gao@siemens.com>
+ *  Chao Zeng <chao.zeng@siemens.com>
+ *
+ * This file is subject to the terms and conditions of the MIT License.  See
+ * COPYING.MIT file in the top-level directory.
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <gpiod.h>
+#include <string.h>
+#include "sp339e.h"
+#include "switchserialmode.h"
+
+void gpio_set(const uint8_t *line_name, uint32_t value)
+{
+    struct gpiod_line *line;
+
+    line = gpiod_line_find(line_name);
+    if (!line) {
+        printf("Unable to find GPIO line %s\n", line_name);
+        return;
+    }
+
+    if (gpiod_line_request_output(line, "switchserialmode", value) < 0) {
+        perror("gpiod_line_request_output");
+    }
+
+    gpiod_line_release(line);
+    gpiod_line_close_chip(line);
+}
+
+static void sp339e_set_mode(uint32_t mode0, uint32_t mode1)
+{
+    gpio_set("UART0-enable", 1);
+    gpio_set("UART0-mode0", mode0);
+    gpio_set("UART0-mode1", mode1);
+}
+
+static void sp339e_set_termination(uint8_t onoff)
+{
+    gpio_set("UART0-terminate", onoff);
+}
+
+static void sp339e_switch_mode(const uint8_t *mode)
+{
+    if (0 == strcasecmp(mode, "rs232"))
+    {
+        sp339e_set_mode(1, 0);
+    }
+    else if(0 == strcasecmp(mode, "rs485"))
+    {
+        sp339e_set_mode(0, 1);
+    }
+    else if(0 == strcasecmp(mode, "rs422"))
+    {
+        sp339e_set_mode(1, 1);
+    }
+}
+
+transceiver_ops_t sp339eOps = {
```

Comment:
Keep the same naming style, `sp339e_ops`?

Reply:
unify the naming rules already